src/components/redux/State.js

```diff
@@ -37,25 +37,28 @@ const store = {
   _callSubscriber() {
     console.log('State changed');
   },
-  addPost(){
-    let newPost = {
-      id: 5,
-      message: this._state.ProfilePage.newPostText , 
+  
+  subscribe(observer) {
+    this._callSubscriber = observer;
+  } ,
 
-      likesCount: 0
-    };
-    this._state.ProfilePage.postsData.push(newPost);
-    this._state.ProfilePage.newPostText = ('');
-    this._callSubscriber(this._state);
-  },
-  updateNewpostText(newText) {
-    this._state.ProfilePage.newPostText = newText;
+  dispatch(action)  {
+    if (action.type === 'ADD-POST') {
+      let newPost = {
+        id: 5,
+        message: this._state.ProfilePage.newPostText , 
+  
+        likesCount: 0
+      };
+      this._state.ProfilePage.postsData.push(newPost);
+      this._state.ProfilePage.newPostText = ('');
       this._callSubscriber(this._state);
-  },
 
-  subscribe(observer) {
-    this._callSubscriber = observer;
-  } 
+    } else if (action.type === 'UPDATE-NEW-POST-TEXT') {
```

Comment:
Consider importing an ACTION_TYPES object and getting this type from there.

Reply:
done